src/components/inputs/text-input/text-input.js

```diff
@@ -7,20 +7,20 @@ import { getInputStyles } from '../styles';
 import { getConstraintSyles } from '../../constraints/horizontal';
 import throwDeprecationWarning from '../../../utils/warn-deprecated-prop';
 
-class TextInput extends React.PureComponent {
-  getInputProps = (props = {}) => {
-    const readOnly = props.readOnly || props.isReadOnly;
+const TextInput = props => {
+  function getInputProps(componentProps = {}) {
```

Comment:
Can we move this function outside of the component?

Reply:
Sure, I was thinking it could be useful to be able to call it from the TextInput, like the `isEmpty` fn, but tbh I'm not sure if it would actually ever be used.

Reply:
In that case you need to assign is as a static property

```js
TextInput.getInputProps = getInputProps
```

But I don't see any usage of it. I would just move it outside.

Reply:
Ah, right, I forgot about that!

Anyway, it's moved now.